### PR TITLE
Use mutations in Gatsby Admin

### PR DESCRIPTION
**Pull request by mxstbr**
Admin is alive! :tada: This is adding mutation support from #23335 to Gatsby Admin (#23291) in order to make it possible to add and remove gatsby plugins—including editing the gatsby-config.js and installing/uninstalling the dependency!! :fire:

### CODE REVIEW

File: packages/gatsby-recipes/src/providers/gatsby/plugin.js [hunk 1]
```diff
@@ -122,6 +122,7 @@ const read = async ({ root }, id) => {
 }
 
 const destroy = async ({ root }, { name }) => {
+  console.log(`DESTROY`, name)
```
Comment (LekoArts) [inline packages/gatsby-recipes/src/providers/gatsby/plugin.js:125]: Stray console.log

File: packages/gatsby-recipes/src/providers/gatsby/plugin.js [hunk 2]
```diff
@@ -134,6 +135,7 @@ const destroy = async ({ root }, { name }) => {
     plugins: [addPlugins.plugin],
     configFile: false,
   })
+  console.log(code)
```
Comment (LekoArts) [inline packages/gatsby-recipes/src/providers/gatsby/plugin.js:138]: Stray console.log

File: packages/gatsby-admin/src/pages/index.tsx [hunk 3]
```diff
@@ -1,5 +1,72 @@
...
+import { useQuery, useMutation } from "urql"
+
+const InstallInput = () => {
+  const [value, setValue] = React.useState("")
+
+  const [_, installGatbyPlugin] = useMutation(`
+    mutation installGatsbyPlugin($name: String!) {
+      createNpmPackage(npmPackage: {
+        name: $name,
+        dependencyType: "development"
```
Comment (LekoArts) [inline packages/gatsby-admin/src/pages/index.tsx:11]: Just for my own curiosity: What does `dependencyType: "development"` mean in this context?

Comment (mxstbr) [reply packages/gatsby-admin/src/pages/index.tsx:11]: The npm package gets added to the `devDependencies`!

Comment (LekoArts) [reply packages/gatsby-admin/src/pages/index.tsx:11]: Why tho? You typically add your plugin's npm package to `dependencies` 🤔

Comment (mxstbr) [reply packages/gatsby-admin/src/pages/index.tsx:11]: To me, anything that isn't shipped to the browser is a devDep—I don't really care either way though, we can bikeshed that 😬🤷‍♂️

Comment (LekoArts) [reply packages/gatsby-admin/src/pages/index.tsx:11]: Our docs use `dependencies` so let’s roll with that 😀

File: packages/gatsby-admin/src/pages/index.tsx [hunk 4]
```diff
@@ -1,5 +1,72 @@
 import React from "react"
-import { useQuery } from "urql"
+import { useQuery, useMutation } from "urql"
+
+const InstallInput = () => {
```
Comment (ascorbic) [inline packages/gatsby-admin/src/pages/index.tsx:4]: The linter is complaining about missing return values, which is why it's failing tests. Type them as `React.FC` and you'll get that for free

File: packages/gatsby-admin/src/pages/index.tsx [hunk 5]
```diff
@@ -1,5 +1,72 @@
...
+        id
+        name
+      }
+    }
+  `)
+
+  return <button onClick={() => deleteGatsbyPlugin({ name })}>X</button>
+}
 
 export default (): React.ReactElement => {
```
Comment (ascorbic) [inline packages/gatsby-admin/src/pages/index.tsx:71]: If you give this a name and export that as default afterwards it should show up properly in devtools. As above, typing it as React.FC will give correct return and props values

File: packages/gatsby-admin/src/pages/index.tsx [hunk 6]
```diff
@@ -1,5 +1,72 @@
...
+          onChange={evt => setValue(evt.target.value)}
+          type="text"
+          placeholder="gatsby-plugin-cool"
+        />
+      </label>
+    </form>
+  )
+}
+
+const DestroyButton = ({ name }) => {
```
Comment (ascorbic) [inline packages/gatsby-admin/src/pages/index.tsx:47]: You'll need to use an interface to type the props here. Make it `React.FC<ButtonProps>` or similar

**Comment by ascorbic** [comment]
I've re-run the cloud tests and they now pass

**Comment by mxstbr** [comment]
Thanks for the detailed review @ascorbic! :pray:

**Comment by laurieontech** [comment]
Worked for me locally!

**Comment by mxstbr** [comment]
It _kinda_ worked for @laurieontech locally :grimacing: Only requires editing a gitignored bundle file and copying some node_modules...

**Comment by laurieontech** [comment]
Haha it counts as working as long as the README tells you how to make it work...right?

**Pull request merged**